docs/develop/testing-cloe.rst

```diff
@@ -0,0 +1,180 @@
+Testing Cloe
+============
+
+ .. note::
+    Please ensure you :doc:`understand<../usage/understanding-cloe-packages>`
+    how Cloe packages work before reading this section.
+
+Cloe packages need to be tested in isolation as well as together with other
+packages. To this end, packages may include unit tests, example executables,
+and integration tests.
+
+Unit Tests
+----------
+
+With the help of CMake and the Google Test framework, unit tests (written in
+C++) can be run within each package directory with::
+
+    make test
+
+Note that this requires a local build (``make all``) to work. Available unit
+tests are automatically executed after each package was built, if the Conan
+recipe option ``test`` is set to ``True`` (default: ``True``). This means,
+a sucessful build usually includes sucessful test execution, such as when we
+run::
+
+    make package
+
+or::
+
+    make all
+
+It's therefore usually not necessary to explicitely run the unit tests, unless
+there is a failing test.
+
+To understand what's going on, it's recommended to check out the targets as
+defined in the ``Makefile.package`` file in the repository root.
+
+When tests are executed automatically, CMake collates available tests and runs
+them with its own command line output. If a test fails, all you see is that
+a test failed, but you won't see *why* it failed, nor will you directly be able
+to debug the test. In order to that, you need to find the test executable that
+CMake executed and run that yourself.
```

Comment:
Here, we should mention that it might be necessary to run the unit test executable from a cloe shell. Otherwise, dynamically linked libraries may not be found by the linker.

Reply:
done